### ISSUE TITLE
[Question] Full List of Built In Fixtures

### ISSUE THREAD

**Issue by davidrosevear** [microsoft/playwright#20831]
Hi,

I’ve tried looking in multiple places, but cannot seem to find anywhere that actually provides the full list of built in fixtures, only alludes to it.

https://playwright.dev/docs/test-fixtures#built-in-fixtures provides a small “list of the pre-defined fixtures that you are likely to use most of the time”, but doesn’t then say how to find a full list.

https://playwright.dev/docs/test-fixtures#overriding-fixtures shows a couple of testOption ones, showing potential that perhaps all TestOption properties are fixtures. After reading https://playwright.dev/docs/api/class-test#test-use and seeing a PW video on fixtures, looking through some code, and seeing the intellisense and typing on tests and extend, this seems to be correct. But wanted to validate.

So long story short, my question is whether there’s an existing list or indication of what the full list of built in fixtures are, or even if not could you confirm whether the ones showing on the fixtures page along with the full TestOption object properties comprehensively make up the list of built-ins?

**Comment by ltsuda** [comment]
Adding pytest fixtures CLI option for use case example:

`pytest --fixtures`
`pytest --fixtures-per-test`

https://docs.pytest.org/en/stable/builtin.html
https://docs.pytest.org/en/stable/reference/reference.html#command-line-flags

**Comment by aslushnikov** [comment]
> After reading https://playwright.dev/docs/api/class-test#test-use and seeing a PW video on fixtures, looking through some code, and seeing the intellisense and typing on tests and extend, this seems to be correct. But wanted to validate.

@davidrosevear technically you're right. All options are fixtures, but please perceive them differently, since "options" are for configuration rather than runtime exploitation. So the list of fixtures on https://playwright.dev/docs/test-fixtures#built-in-fixtures is complete. (in a sense we'd like you to think about them). 

> Adding pytest fixtures CLI option for use case example

@ltsuda thank you for the reference! We discussed this on the team meeting and decided to keep things as they are for now.

**Comment by davidrosevear** [comment]
@aslushnikov Thank you for the confirmation and clarification! I guess to clarify a bit more, what do you mean by “runtime exploitation”? It seems in the section in https://playwright.dev/docs/test-fixtures#overriding-fixtures uses/exploits them as examples, so thought the clarification might be useful.
